<a href="https://colab.research.google.com/github/Nurulhasanah2/Data-Mining/blob/main/Naive_Bayes.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Naive Bayes


1. Load dataset

In [ ]:
import numpy as np
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.naive_bayes import GaussianNB
from sklearn.metrics import accuracy_score
import matplotlib.pyplot as plt

2. Input Data dan pengecekan missing value

In [ ]:
df = pd.read_excel("iris.xlsx")
df

SepalLength  SepalWidth  PetalLength  PetalWidth           Label
0            5.1         3.5          1.4         0.2     Iris-setosa
1            4.9         3.0          1.4         0.2     Iris-setosa
2            4.7         3.2          1.3         0.2     Iris-setosa
3            4.6         3.1          1.5         0.2     Iris-setosa
4            5.0         3.6          1.4         0.2     Iris-setosa
..           ...         ...          ...         ...             ...
145          6.7         3.0          5.2         2.3  Iris-virginica
146          6.3         2.5          5.0         1.9  Iris-virginica
147          6.5         3.0          5.2         2.0  Iris-virginica
148          6.2         3.4          5.4         2.3  Iris-virginica
149          5.9         3.0          5.1         1.8  Iris-virginica

[150 rows x 5 columns]

In [ ]:
df.info() #informasi data

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   SepalLength  150 non-null    float64
 1   SepalWidth   150 non-null    float64
 2   PetalLength  150 non-null    float64
 3   PetalWidth   150 non-null    float64
 4   Label        150 non-null    object 
dtypes: float64(4), object(1)
memory usage: 6.0+ KB


In [ ]:
df.isna().sum() #pengecekan missing value

SepalLength    0
SepalWidth     0
PetalLength    0
PetalWidth     0
Label          0
dtype: int64

Berdasarkan Output diatas dapat diketahui bahwa data tidak memiliki missing values

3.Split Data

In [ ]:
x = df.drop(['Label'], axis=1) #Variabelindependen
x

y = df['Label'] #Variabel Dependen
y

0         Iris-setosa
1         Iris-setosa
2         Iris-setosa
3         Iris-setosa
4         Iris-setosa
            ...      
145    Iris-virginica
146    Iris-virginica
147    Iris-virginica
148    Iris-virginica
149    Iris-virginica
Name: Label, Length: 150, dtype: object

Pada tahap ini menentukan variabel independennya yaitu variabel SepalLength, SepalWidth, PetalLength, dan PetalWidth sedangkan variabel dependennya yaitu variabel Label

In [ ]:
from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test = train_test_split( x, y, test_size=0.2, random_state=4)
print ('Train set:', x_train.shape,  y_train.shape)
print ('Test set:', x_test.shape,  y_test.shape)

Train set: (120, 4) (120,)
Test set: (30, 4) (30,)


Dilakukan pembagian antara data training dan data testing sebesar 80% data training dan 20% data testing

4. Implementasi Naive Bayes

In [ ]:
model = GaussianNB()
model.fit(x_train,y_train)

GaussianNB()

In [ ]:
y_pred = model.predict(x_test)

In [ ]:
print(y_pred[0:5])
print(y_test[0:5])

['Iris-virginica' 'Iris-setosa' 'Iris-virginica' 'Iris-virginica'
 'Iris-virginica']
128    Iris-virginica
18        Iris-setosa
130    Iris-virginica
105    Iris-virginica
107    Iris-virginica
Name: Label, dtype: object


5. Perhitungan Akurasi

In [ ]:
from sklearn import metrics
print("Test set Accuracy: ", metrics.accuracy_score(y_pred, y_test))

Test set Accuracy:  0.9666666666666667


Hasil Akurasi penggunaan metode Naive Bayes dalam kasus data ini adalah sebesar 0.96666667